pkg/server/skydb/pq/device.go

```diff
@@ -24,16 +24,46 @@ import (
 )
 
 func (c *conn) GetDevice(id string, device *skydb.Device) error {
-	builder := psql.Select("type", "token", "user_id", "last_registered_at").
+	builder := psql.Select("type", "token", "user_id", "topic", "last_registered_at").
 		From(c.tableName("_device")).
-		Where("id = ?", id)
+		Where("id = ? AND topic IS NOT NULL", id)
```

Comment:
in the following cases:
* pusher that does not require a topic to send a notification (do they exists?)
* users who migrated from older version of skygear-server to new version but they keep using cert-based APNS pusher

`topic IS NOT NULL` will never return a device because there is no topic associated with the device.